src/compiler/types.ts

```diff
@@ -3907,6 +3918,8 @@ namespace ts {
         resolvedModule: ResolvedModuleFull | undefined;
         /* @internal */
         failedLookupLocations: string[];
+        /*@internal*/
+        isInvalidated?: boolean;

```

Comment:
To clarify, I meant to use a *different* type than `ResolvedModuleWithFailedLookupLocations` for mutable purposes, not just to add a new field but mark the others readonly.